### ISSUE TITLE
STM32F103 Custom Board Clock Config Error

### ISSUE THREAD

**Issue by lainslie** [zephyrproject-rtos/zephyr#44822]
Hi there,

I recently started working with a custom stm32f103c8t6 pcb. I seem to be having trouble with setting up the device tree correctly for the system clock. I was able to successfully flash the pcb, with blinking led sample, through stm32cubeide with the following clock configuration:

![image](https://user-images.githubusercontent.com/58678981/163141347-7738f69b-c352-4df4-bf4c-323d5dbf53d3.png)

With this config, I attempted to replicate it in my {boardname}.dts file in the following manner:

![image](https://user-images.githubusercontent.com/58678981/163141685-4b31acca-5406-4aea-82cb-77f6f297207f.png)

I have been able to verify that in fact my program is being programmed onto the microcontroller, however, when the GPIO pin is written to, nothing occurs. Further to this, if k_msleep is used, the program crashes, which pointed in the direction of a clock issue? Any help would be greatly appreciated.

I've used both the zephyrsdk and gnuarm toolchain just to exhaust all options
Windows 10

additional details:
Defconfig -
![image](https://user-images.githubusercontent.com/58678981/163142463-ab767898-84aa-41f2-82e0-07d1324bd9a1.png)

**Comment by lainslie** [comment]
Hi @FRASTM, 

Thanks for your prompt reply. Unfortunately that does not make a difference. I am very unsure what could possibly be wrong.

**Comment by FRASTM** [comment]
Setting the systemClock from the HSI 8MHz is not usual. But I do not see any pb with the nucleo_f103rb running the samples/basic/blinky
(note that the stm32Cube calls LL_RCC_HSI_SetCalibTrimming(16) before enabling the HSI clock.)

**Comment by erwango** [comment]
@lainslie Is your problem still present ?

If yes, can you try to replicate this configuration and see if solving the issue?
```
&clk_hsi {
	//hse-bypass;
	//clock-frequency = <DT_FREQ_M(8)>; /* STLink 8MHz clock */
	status = "okay";
};

&pll {
	mul = <9>;
	clocks = <&clk_hsi>;
	status = "okay";
};

&rcc {
	clocks = <&pll>;
	clock-frequency = <DT_FREQ_M(72)>;
	ahb-prescaler = <1>;
	apb1-prescaler = <2>;
	apb2-prescaler = <1>;
};
```

Also, regarding the code used, can you try to use `sample/basic/blinky` (if you have a led on your board) and report the status ?

**Comment by lainslie** [comment]
Hi there, 

Thanks for all the replies. I ended up finding the issue, and it was not a clock configuration issue, meaning the original clock config that I posted did in fact work. The issue was due to the fact that the LED Pin that was being used was also used for JTAG programming whereas I only required SWD programming. 

It was fixed by adding this to my custom .dts file:

&pinctrl {
	swj-cfg = "jtag-disable";
};
